core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTextAggregator.java

```diff
@@ -0,0 +1,271 @@
+                                lastTrieSize = newTrieSize;
+                            }
+                            previous.clear();
+                            previous.copyChars(termAtt);
+                            BytesRef bytes = previous.get();
+                            if(inDocTerms.add(bytes) >= 0){
+                                if (includeExclude == null || includeExclude.accept(bytes)) {
+                                    long bucketOrdinal = bucketOrds.add(bytes);
+                                    if (bucketOrdinal < 0) { // already seen
+                                        bucketOrdinal = - 1 - bucketOrdinal;
+                                        collectExistingBucket(sub, doc, bucketOrdinal);
+                                    } else {
+                                        collectBucket(sub, doc, bucketOrdinal);
+                                    }
+                                }                            
+                            }                            
+                        }
+                    }
+                    finally{
+                        Releasables.close(inDocTerms);
+                    }
+                }
+                finally{
+                    ts.close();
+                }
+            }
+            
+            private void loadFromSource(int doc, long bucket, String fieldName) throws IOException {
+                final DocumentStoredFieldVisitor visitor = new DocumentStoredFieldVisitor("_uid");
+                ctx.reader().document(doc, visitor);
+                Document storedDoc = visitor.getDocument();
+                // Inefficient - we have the Lucene doc here but all the doc-source-loading code assumes we retrieve by elasticsearch 
+                // type+id so we have to run it through the existing "get" service which will incure another Lucene read. 
+                
+                //BUT TODO - talk to MVG about existing (fetch?) apis that allow loading of docs using a Lucene doc id and reader
+                Uid uid = Uid.createUid(storedDoc.get("_uid"));
+                String [] gFields= {"content"};
+                GetResult getResult = shardGetService.get(uid.type(), uid.id(), gFields, false, Versions.MATCH_ANY, VersionType.INTERNAL, 
+                        FetchSourceContext.FETCH_SOURCE);
+                
+                
+                MapperService mapperService = context.indexShard().mapperService();
+                MappedFieldType fieldType = mapperService.fullName(fieldName);
+                if(fieldType == null){
+                    throw new IllegalArgumentException("Aggregation [" + name + "] cannot process field ["+fieldName
+                            +"] since it is not present");                    
+                }
+                Analyzer analyzer = fieldType.indexAnalyzer();
+                Object fieldValue = getResult.getSource().get(fieldName);
```

Comment:
HighlightUtils looks to have a dependency on FetchSubPhase.HitContext (and my agg is only in search phase) but the relevant part of HighlightUtils that uses SourceLookup solves the "TODO" comment I had about retrieving docs via Lucene segment and doc ID rather than existing get-by-elasticsearch-ID nonsense.